util/test_timeline.py

```diff
@@ -39,38 +39,69 @@ def load_timeline(timeline):
             if not sync_match:
                 continue
 
-            entry["regex"] = sync_match.group(1).replace(":", "\|")
             entry["branch"] = 0
 
+            ability_match = e_tools.is_tl_line_cast(sync_match.group(1))
+            if ability_match:
+                entry["regex"] = "2[12]\|[^\|]*\|........\|{}\|{}\|".format(
+                    ability_match.group("source"), ability_match.group("id")
+                )
             # Special casing on syncs
             entry["special_type"] = False
 
             begincast_match = e_tools.is_tl_line_begincast(sync_match.group(1))
             if begincast_match:
                 entry["special_type"] = "begincast"
                 entry["special_line"] = "20"
-                entry["cast_id"] = begincast_match.group(1)
-                entry["caster_name"] = begincast_match.group(2)
+                entry["cast_id"] = begincast_match.group("id")
+                entry["caster_name"] = begincast_match.group("source")
```

Comment:
Did you verify that all of these log types: cast, begincast, buff, log, adds, headmarker all match properly? Mostly just looking for "have each of these conditional blocks been run at least once and seem to work" level of testing.

Reply:
Verified now. SO much nonsense.